### ISSUE TITLE
HPACK: Use Netty's version of HPACK

### ISSUE THREAD

**Issue by ktoso** [akka/akka-http#304]
It was imported from the twitter one, however is higher performance (less allocations).
See here: https://github.com/netty/netty/tree/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack

To be decided if we shade or can depend on it.

More info: https://github.com/twitter/hpack/pull/32

**Comment by jrudolph** [comment]
This would require to pull into a considerable part of netty infrastructure, most importantly we would need to support ByteBuf. We'll need to see if that's really necessary or if we can import and fix twitter HPACK to our codebase similar to what netty has done to adapt it to the needs of its infrastructure.

**Comment by ktoso** [comment]
Yup agreed, as discussed IRL earlier. I think we'll end up having to have a copy of it and get it possibly as copy-free as possible.
